pysamstats/test/test_io.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+import logging
+
+from pysam import Samfile
+from pysamstats import load_coverage, stat_coverage
+
+from pysamstats.io import write_hdf5
+import tables
+
+logger = logging.getLogger(__name__)
+debug = logger.debug
+
+
+def write_hdf5_dtype(arg):
+
+    # testing auto dtype determination.
+    dtype, alignment, result, label = arg
+    import tempfile
+
+    # use auto
+    tmp = tempfile.mktemp(suffix=".h5")
+    write_hdf5("coverage", tmp, alignment, chrom=label, dtype=dtype)
+    with tables.open_file(tmp, mode="r") as h5file:
+        assert result == h5file.root.data.dtype["chrom"].itemsize
+
+
+def test_write_hdf5():
+
+    contig_label = "AS2_scf7180000696055"
+    bampath = "fixture/longcontignames.bam"
+
+    dtypes = [None, {"chrom": "a20"}, {"chrom": "a20"}]
+    alignments = [Samfile(bampath), Samfile(bampath), bampath]
+    results = [len(contig_label), 20, 20]
+    labels = [contig_label, contig_label, contig_label]
+
+    for arg in zip(dtypes, alignments, results, labels):
+        yield (write_hdf5_dtype, arg)
```

Comment:
No need to yield here, you can just call write_hdf5_dtype(arg).

Reply:
I used `yield` as that was the only way I could get it to count as 3 tests rather than 1. Have removed, 1 test is fine!

Reply:
Ah OK. Yes 1 test is fine I think.